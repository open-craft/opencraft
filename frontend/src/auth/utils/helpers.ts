import { V2Api } from 'global/api';

export const checkAuthAndRefreshToken = async () => {
  const refreshToken = localStorage.getItem('token_refresh');
  let response = null;

  if (refreshToken) {
    try {
      response = await V2Api.authRefreshCreate({
        data: { refresh: refreshToken }
      });
    } catch (e) {
      return false;
    }
  }

  if (response) {
    localStorage.setItem('token_access', response.access);
    return true;
  }

  return false;
};
