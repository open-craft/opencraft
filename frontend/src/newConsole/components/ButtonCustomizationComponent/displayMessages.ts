const messages = {
  switchTooltipMessage: {
    defaultMessage:
      'You need to set all options before activating the button style override.',
    description: ''
  }
};

export default messages;
