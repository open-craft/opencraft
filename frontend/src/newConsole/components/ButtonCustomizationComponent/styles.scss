.button-customization-container {
  margin-bottom: 3rem;
}

.button-customization-name {
  width: 100%;
  height: 29px;
  font-family: Chivo;
  font-size: 24px;
  font-weight: normal;
  font-stretch: normal;
  font-style: normal;
  line-height: normal;
  letter-spacing: 0.72px;
  color: #0f1f24;
  display: inline-flex;
  align-items: center;

  .custom-switch {
    margin-left: 10px;
    display: inline-flex;
  }

  span.info-icon {
    margin-left: 10px;
    display: inline-flex;

    .custom-switch {
      margin-left: 0;
    }
  }
}
