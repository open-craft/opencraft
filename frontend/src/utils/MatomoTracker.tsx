import React, { ReactNode } from 'react';
import {
  createInstance,
  MatomoProvider,
  useMatomo
} from '@datapunt/matomo-tracker-react';
import {
  MATOMO_ALIAS_DOMAIN,
  MATOMO_BASE_URL,
  MATOMO_MY_DOMAIN,
  MATOMO_SITE_ID
} from '../global/constants';

interface MatomoTrackerProps {
  children: ReactNode;
}

interface MatomoAdditionalConfiguration {
  [propName: string]: any;
}

interface MatomoConfiguration {
  urlBase: string;
  siteId: number;
  configurations?: MatomoAdditionalConfiguration;
}

interface MatomoUserIdTrackerProps {
  userId: string;
}

/*
 * This wrapper component initializes the Matomo tracking by wrapping the children
 * elements, if the tracking is enabled by providing a valid value for the 'MATOMO_BASE_URL'
 * constant. It allows tracking page views, events and the user id, when available.
 *
 * The 'MATOMO_ALIAS_DOMAIN' constant allows the cross-domain linking of another site with
 * this site so that they are treated as the same site in Matomo.
 */
export const MatomoTracker: React.FC<MatomoTrackerProps> = (
  props: MatomoTrackerProps
) => {
  let element = props.children;
  if (MATOMO_BASE_URL) {
    const matomoConfiguration: MatomoConfiguration = {
      urlBase: MATOMO_BASE_URL,
      siteId: parseInt(MATOMO_SITE_ID, 10)
    };
    if (MATOMO_ALIAS_DOMAIN) {
      const myDomain = `*.${MATOMO_MY_DOMAIN}`;
      const aliasDomain = `*.${MATOMO_ALIAS_DOMAIN}`;
      matomoConfiguration.configurations = {
        enableCrossDomainLinking: true,
        setDomains: [myDomain, aliasDomain]
      };
    }
    element = (
      <MatomoProvider value={createInstance(matomoConfiguration)}>
        {props.children}
      </MatomoProvider>
    );
  }
  return <>{element}</>;
};

/**
 * This is a component passes the provided user id to Matomo if Matomo tracking is enabled.
 * The 'pushInstruction' function exposed by the 'useMatomo()' hook is used to do this.
 *
 * This component is needed as all the components in the registration process are classes and hence
 * the 'yseMatomo' hook cannot be directly used in those components.
 */
export const MatomoUserIdTracker: React.FC<MatomoUserIdTrackerProps> = (
  props: MatomoUserIdTrackerProps
) => {
  const { pushInstruction } = useMatomo();
  if (MATOMO_BASE_URL && props.userId) {
    pushInstruction('setUserId', props.userId);
  }
  return null;
};
