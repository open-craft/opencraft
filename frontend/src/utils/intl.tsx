import React from "react";
import { FormattedMessage, MessageDescriptor } from "react-intl";

type MessageMap = { [k: string]: Omit<MessageDescriptor, 'id'> };

interface Props<T, K extends keyof T> {
    id: K;
    messages: T;
    values?: any;
}


export const WrappedMessage = (props: Props<MessageMap, string>) => {
    const {messages, id, ...values} = props;
    return (
        <FormattedMessage  {...messages[id]} id={id} {...values}/>
    );
};
