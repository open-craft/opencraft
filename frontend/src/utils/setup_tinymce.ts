import 'tinymce/tinymce';
import 'tinymce/themes/silver';
import 'tinymce/skins/ui/oxide/skin.min.css';
import 'tinymce/skins/ui/oxide/content.min.css';
import 'tinymce/plugins/advlist';
import 'tinymce/plugins/autolink';
import 'tinymce/plugins/lists';
import 'tinymce/plugins/link';
import 'tinymce/plugins/image';
import 'tinymce/plugins/charmap';
import 'tinymce/plugins/print';
import 'tinymce/plugins/preview';
import 'tinymce/plugins/anchor';
import 'tinymce/plugins/searchreplace';
import 'tinymce/plugins/visualblocks';
import 'tinymce/plugins/code';
import 'tinymce/plugins/fullscreen';
import 'tinymce/plugins/insertdatetime';
import 'tinymce/plugins/media';
import 'tinymce/plugins/table';
import 'tinymce/plugins/paste';
import 'tinymce/plugins/help';
import 'tinymce/plugins/wordcount';
