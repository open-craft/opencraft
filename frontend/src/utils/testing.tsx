import { createMemoryHistory } from 'history';
import * as React from 'react';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router';
import * as renderer from 'react-test-renderer';
import { applyMiddleware, createStore } from 'redux';
import thunk from 'redux-thunk';
import { createRootReducer } from '../global/reducers';

export const setupComponentForTesting = (
  reactContent: JSX.Element,
  storeContents = {}
) => {
  const middleware = applyMiddleware(thunk);

  const store = createStore(
    createRootReducer(createMemoryHistory()),
    storeContents,
    middleware
  );

  return renderer.create(
    <IntlProvider textComponent={React.Fragment} locale="en">
      <Provider store={store}>
        <MemoryRouter>{reactContent}</MemoryRouter>
      </Provider>
    </IntlProvider>
  );
};
