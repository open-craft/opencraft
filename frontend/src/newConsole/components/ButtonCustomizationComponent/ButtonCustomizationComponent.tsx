import * as React from 'react';
import { InstanceSettingsModel } from 'console/models';
import { Form, OverlayTrigger, Tooltip } from 'react-bootstrap';
import { WrappedMessage } from 'utils/intl';
import { ThemeSchema } from 'ocim-client';
import messages from './displayMessages';
import './styles.scss';
import { ButtonStyles } from '../ButtonStyles';

interface ButtonCustomizationPageProps {
  buttonName: string;
  externalMessages: any;
  onChangeColor: Function;
  loading: Array<keyof InstanceSettingsModel | 'deployment'>;
  themeData: ThemeSchema;
  initialExpanded?: boolean;
  deploymentToggle?: boolean;
}

export const ButtonCustomizationComponent: React.FC<ButtonCustomizationPageProps> = (
  props: ButtonCustomizationPageProps
) => {
  const { themeData } = props;
  const { loading } = props;
  const buttonFullName = `theme${props.buttonName}Button`;

  const tooltip = (
    <Tooltip id="button-tooltip">
      <WrappedMessage
        messages={props.externalMessages}
        id={`${buttonFullName}Help`}
      />
    </Tooltip>
  );

  const styles = {
    activeStyles: [
      `btn${props.buttonName}Bg`,
      `btn${props.buttonName}Color`,
      `btn${props.buttonName}BorderColor`
    ],
    hoverStyles: [
      `btn${props.buttonName}HoverBg`,
      `btn${props.buttonName}HoverColor`,
      `btn${props.buttonName}HoverBorderColor`
    ]
  };

  let customizationProp = `customize${props.buttonName}Btn`;
  if (props.buttonName === 'Logistration') {
    // Logistration toggle doesn't follow the common naming convention, hence setting it explicitly.
    // See SE-2955 for discussion.
    customizationProp = `customizeLogistrationActionBtn`;
  }

  const allStylesDefined = Object.values(styles)
    .map(category =>
      category.every(style => themeData[style as keyof typeof themeData])
    )
    .every(category => category);
  const customizationEnabled =
    (themeData[customizationProp as keyof typeof themeData] as
      | boolean
      | undefined) || false;

  const switchTooltip = (
    <Tooltip id="button-tooltip">
      <WrappedMessage messages={messages} id="switchTooltipMessage" />
    </Tooltip>
  );
  const switchToggle = (
    <Form.Check
      id={`toggle-${props.buttonName}`}
      type="switch"
      label=""
      checked={customizationEnabled}
      onChange={() => {
        props.onChangeColor(customizationProp, !customizationEnabled);
      }}
      disabled={!allStylesDefined}
    />
  );

  return (
    <div className="button-customization-container">
      <div className="button-customization-name">
        <WrappedMessage messages={props.externalMessages} id={buttonFullName} />
        <OverlayTrigger placement="right" overlay={tooltip}>
          <span className="info-icon">
            <i className="fas fa-info-circle" />
          </span>
        </OverlayTrigger>
        {props.deploymentToggle &&
          (!allStylesDefined ? (
            <OverlayTrigger placement="right" overlay={switchTooltip}>
              <span className="info-icon">{switchToggle}</span>
            </OverlayTrigger>
          ) : (
            switchToggle
          ))}
      </div>
      <div className="button-customization-color-container">
        <ButtonStyles
          buttonName={props.buttonName}
          onChangeColor={props.onChangeColor}
          themeData={themeData}
          loading={loading}
          initialExpanded={props.initialExpanded || false}
        />
      </div>
    </div>
  );
};
