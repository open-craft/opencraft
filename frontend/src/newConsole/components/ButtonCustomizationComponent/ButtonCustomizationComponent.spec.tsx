import React from 'react';
import { setupComponentForTesting } from "utils/testing";
import { ButtonCustomizationComponent } from './ButtonCustomizationComponent';
import messages from "../ButtonsCustomizationPage/displayMessages";

it('renders without crashing', () => {
  const tree = setupComponentForTesting(
    <ButtonCustomizationComponent 
      buttonName="Primary"
      externalMessages={messages}
      onChangeColor={() => {}}
      loading={['draftThemeConfig']}
      themeData={{}}/>).toJSON();
    expect(tree).toMatchSnapshot();
});
