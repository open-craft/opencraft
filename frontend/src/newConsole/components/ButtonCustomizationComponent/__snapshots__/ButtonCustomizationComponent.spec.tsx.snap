// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="button-customization-container"
>
  <div
    className="button-customization-name"
  >
    Primary button
    <span
      className="info-icon"
      onBlur={[Function]}
      onFocus={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
    >
      <i
        className="fas fa-info-circle"
      />
    </span>
  </div>
  <div
    className="button-customization-color-container"
  >
    <div
      className="collapsible-edit-area"
    >
      <div
        className="edit-collapse"
      >
        <a
          className="nav-link"
          href="#"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
        >
          Edit
          <i
            className="fas xs fa-chevron-down"
          />
        </a>
        <span
          className="fill-line"
        />
      </div>
      <div
        aria-expanded={null}
        className="content collapse"
      >
        <div
          className="button-category"
        >
          <div>
            <p
              className="button-state-name"
            >
              Active styles
            </p>
          </div>
          <div>
            <div>
              <div
                className="color-input form-group"
              >
                <label
                  className="form-label"
                >
                  Button color
                </label>
                <div
                  className="align-items-center row"
                >
                  <div
                    className="d-flex align-items-center color-input-field"
                  >
                    <input
                      className="input-field-color form-control"
                      disabled={true}
                      name="btnPrimaryBg"
                      onClick={[Function]}
                      readOnly={true}
                      value=""
                    />
                    <div
                      className="input-field-preview"
                      style={
                        Object {
                          "backgroundColor": "",
                        }
                      }
                    />
                  </div>
                  <button
                    className="reset-value padded flex-grow"
                    onClick={[Function]}
                    type="button"
                  >
                    Reset
                  </button>
                </div>
              </div>
            </div>
            <div>
              <div
                className="color-input form-group"
              >
                <label
                  className="form-label"
                >
                  Button text color
                </label>
                <div
                  className="align-items-center row"
                >
                  <div
                    className="d-flex align-items-center color-input-field"
                  >
                    <input
                      className="input-field-color form-control"
                      disabled={true}
                      name="btnPrimaryColor"
                      onClick={[Function]}
                      readOnly={true}
                      value=""
                    />
                    <div
                      className="input-field-preview"
                      style={
                        Object {
                          "backgroundColor": "",
                        }
                      }
                    />
                  </div>
                  <button
                    className="reset-value padded flex-grow"
                    onClick={[Function]}
                    type="button"
                  >
                    Reset
                  </button>
                </div>
              </div>
            </div>
            <div>
              <div
                className="color-input form-group"
              >
                <label
                  className="form-label"
                >
                  Button border color
                </label>
                <div
                  className="align-items-center row"
                >
                  <div
                    className="d-flex align-items-center color-input-field"
                  >
                    <input
                      className="input-field-color form-control"
                      disabled={true}
                      name="btnPrimaryBorderColor"
                      onClick={[Function]}
                      readOnly={true}
                      value=""
                    />
                    <div
                      className="input-field-preview"
                      style={
                        Object {
                          "backgroundColor": "",
                        }
                      }
                    />
                  </div>
                  <button
                    className="reset-value padded flex-grow"
                    onClick={[Function]}
                    type="button"
                  >
                    Reset
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="button-category"
        >
          <div>
            <p
              className="button-state-name"
            >
              Hover styles
            </p>
          </div>
          <div>
            <div>
              <div
                className="color-input form-group"
              >
                <label
                  className="form-label"
                >
                  Button color
                </label>
                <div
                  className="align-items-center row"
                >
                  <div
                    className="d-flex align-items-center color-input-field"
                  >
                    <input
                      className="input-field-color form-control"
                      disabled={true}
                      name="btnPrimaryHoverBg"
                      onClick={[Function]}
                      readOnly={true}
                      value=""
                    />
                    <div
                      className="input-field-preview"
                      style={
                        Object {
                          "backgroundColor": "",
                        }
                      }
                    />
                  </div>
                  <button
                    className="reset-value padded flex-grow"
                    onClick={[Function]}
                    type="button"
                  >
                    Reset
                  </button>
                </div>
              </div>
            </div>
            <div>
              <div
                className="color-input form-group"
              >
                <label
                  className="form-label"
                >
                  Button text color
                </label>
                <div
                  className="align-items-center row"
                >
                  <div
                    className="d-flex align-items-center color-input-field"
                  >
                    <input
                      className="input-field-color form-control"
                      disabled={true}
                      name="btnPrimaryHoverColor"
                      onClick={[Function]}
                      readOnly={true}
                      value=""
                    />
                    <div
                      className="input-field-preview"
                      style={
                        Object {
                          "backgroundColor": "",
                        }
                      }
                    />
                  </div>
                  <button
                    className="reset-value padded flex-grow"
                    onClick={[Function]}
                    type="button"
                  >
                    Reset
                  </button>
                </div>
              </div>
            </div>
            <div>
              <div
                className="color-input form-group"
              >
                <label
                  className="form-label"
                >
                  Button border color
                </label>
                <div
                  className="align-items-center row"
                >
                  <div
                    className="d-flex align-items-center color-input-field"
                  >
                    <input
                      className="input-field-color form-control"
                      disabled={true}
                      name="btnPrimaryHoverBorderColor"
                      onClick={[Function]}
                      readOnly={true}
                      value=""
                    />
                    <div
                      className="input-field-preview"
                      style={
                        Object {
                          "backgroundColor": "",
                        }
                      }
                    />
                  </div>
                  <button
                    className="reset-value padded flex-grow"
                    onClick={[Function]}
                    type="button"
                  >
                    Reset
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
